### PR TITLE
feature(admin/functions): add payment process tasks

### LINKED ISSUE #313: Add another download csv button to recipients view
To make the payments in the Orange Money web interface, a CSV with all the registered recipients needs to be uploaded. To simplify this process, it should be possible to always download the up-to-date CSV from the admin tool.

The current button “DOWNLOAD ORANGE MONEY CSV” should be renamed to “DOWNLOAD PAYMENTS CSV”, and the new button should be called "DOWNLOAD RECIPIENTS CSV". Both should trigger the same function call. To create the CSV, I would extend the current function and pass a parameter that defines which type of CSV should be created. 

Similar to the payments CSV, the content should only contain recipients with state `Active` and `Designated`, see https://github.com/socialincome-san/public/blob/929f1b6a278e46af91f7a2ebc16b5bfa60649666/functions/src/etl/OrangeMoneyCSVCreator.ts#L19

#### Format of the new payments CSV (one entry per recipient)

```csv
Mobile Number*,Unique Code*,User Type*
33333333,1,subscriber
55555555,2,subscriber
44444444,3,subscriber
```

### PR DESCRIPTION
**Pull request by mkue**
Resolves #313 

(Steps 1-3 in #286)

### CODE REVIEW

**Review by ssandino — APPROVED** [review]

File: shared/src/types/admin/Payment.ts [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
+export enum PaymentStatus {
+	Created = 'created',
+	Paid = 'paid',
+	Confirmed = 'confirmed',
+	Contested = 'contested',
+	Failed = 'failed',
+	Other = 'other',
+}
+
 export type Payment = {
```
Comment (mkue) [inline shared/src/types/admin/Payment.ts:14]: @MDikkii: I slightly updated the fields on the payment document. You might have to update this on your end as well (payment status values are a little different, some fields are removed, and a `comments` field was added).

File: functions/package.json [hunk 2]
```diff
@@ -21,7 +21,6 @@
 		"@playwright/test": "^1.30.0",
 		"@types/imap-simple": "^4.2.5",
 		"@types/jest": "^29.4.0",
-		"@types/deep-equal": "^1.0.1",
```
Comment (andrashee) [inline functions/package.json:24]: 👍

File: shared/src/types/admin/Payment.ts (hunk 1 above)
Comment (andrashee) [reply shared/src/types/admin/Payment.ts:14]: @mkue what does that mean for the existing values (e.g. payment status) in firestore? Will they be displayed correctly in the admin?

**Review by andrashee — APPROVED** [review]
Awesome stuff @mkue!
Looking forward to trying it out

File: shared/src/types/admin/Payment.ts (hunk 1 above)
Comment (mkue) [reply shared/src/types/admin/Payment.ts:14]: yes, all payments are currently set to 'confirmed', so it shouldn't be a problem at all.

Comment (mkue) [reply shared/src/types/admin/Payment.ts:14]: ok, there are 10 entries that need to be updated. I'll do it by hand quickly

Comment (MDikkii) [reply shared/src/types/admin/Payment.ts:14]: Don't we want to keep some timestamp for changes (confirmation / contest timestamp)? 

Contest reason probably can go to `comments`, right?

Comment (mkue) [reply shared/src/types/admin/Payment.ts:14]: Yes, I would say so. The changes are now automatically logged to a subcollection, see #323. So the information is there, and if needed for some reason, we can also add the fields back.

**Pull request merged**